components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -342,6 +343,29 @@ static esp_err_t esp_websocket_client_destroy_config(esp_websocket_client_handle
     return ESP_OK;
 }
 
+static void destroy_selected(esp_websocket_client_handle_t client)
+{
+    if (client->selected_for_destroying == false) {
+        return;
+    }
+    if (client->event_handle) {
+        esp_event_loop_delete(client->event_handle);
+    }
+    if (client->if_name) {
+        free(client->if_name);
+    }
+    esp_websocket_client_destroy_config(client);
+    esp_transport_list_destroy(client->transport_list);
+    vQueueDelete(client->lock);
+    free(client->tx_buffer);
+    free(client->rx_buffer);
+    if (client->status_bits) {
+        vEventGroupDelete(client->status_bits);
+    }
+    free(client);
+    client = NULL;
```

Comment:
Question: Could this part be reused in the standard `esp_websocket_client_destroy()` ?